modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/planner/datatypes/NumericBinaryOperationsTypeCoercionTest.java

```diff
@@ -0,0 +1,813 @@
+ */
+public class NumericBinaryOperationsTypeCoercionTest extends BaseTypeCoercionTest {
+
+    // No any type changes for `addition` operation from planner perspective.
+    @ParameterizedTest
+    @MethodSource("allPairs")
+    public void additionOp(NumericPair pair) throws Exception {
+        IgniteSchema schema = createSchemaWithTwoColumnTable(pair.first(), pair.second());
+
+        Matcher<RexNode> first = ofTypeWithoutCast(pair.first());
+        Matcher<RexNode> second = ofTypeWithoutCast(pair.second());
+
+        assertPlan("SELECT c1 + c2 FROM t", schema, operandMatcher(first, second)::matches, List.of());
+        assertPlan("SELECT c2 + c1 FROM t", schema, operandMatcher(second, first)::matches, List.of());
+    }
+
+    // No any type changes for `subtraction` operation from planner perspective.
+    @ParameterizedTest
+    @MethodSource("allPairs")
+    public void subtractionOp(NumericPair pair) throws Exception {
+        IgniteSchema schema = createSchemaWithTwoColumnTable(pair.first(), pair.second());
+
+        Matcher<RexNode> first = ofTypeWithoutCast(pair.first());
+        Matcher<RexNode> second = ofTypeWithoutCast(pair.second());
+
+        assertPlan("SELECT c1 - c2 FROM t", schema, operandMatcher(first, second)::matches, List.of());
+        assertPlan("SELECT c2 - c1 FROM t", schema, operandMatcher(second, first)::matches, List.of());
+    }
+
+    // No any type changes for `division` operation from planner perspective.
+    @ParameterizedTest
+    @MethodSource("allPairs")
+    public void divisionOp(NumericPair pair) throws Exception {
+        IgniteSchema schema = createSchemaWithTwoColumnTable(pair.first(), pair.second());
+
+        Matcher<RexNode> first = ofTypeWithoutCast(pair.first());
+        Matcher<RexNode> second = ofTypeWithoutCast(pair.second());
+
+        assertPlan("SELECT c1 / c2 FROM t", schema, operandMatcher(first, second)::matches, List.of());
+        assertPlan("SELECT c2 / c1 FROM t", schema, operandMatcher(second, first)::matches, List.of());
+    }
+
+    // No any type changes for `multiplication` operation from planner perspective.
+    @ParameterizedTest
+    @MethodSource("allPairs")
+    public void multiplicationOp(NumericPair pair) throws Exception {
+        IgniteSchema schema = createSchemaWithTwoColumnTable(pair.first(), pair.second());
+
+        Matcher<RexNode> first = ofTypeWithoutCast(pair.first());
+        Matcher<RexNode> second = ofTypeWithoutCast(pair.second());
+
+        assertPlan("SELECT c1 * c2 FROM t", schema, operandMatcher(first, second)::matches, List.of());
+        assertPlan("SELECT c2 * c1 FROM t", schema, operandMatcher(second, first)::matches, List.of());
+    }
+
+    //Have the following casts for modulo operation:
+    // REAL datatype always casts to DECIMAL(14,7)
+    // DOUBLE datatype always casts to DECIMAL(30,15)
+    // Any other types with no any changes
+    @ParameterizedTest
+    @MethodSource("moduloArgs")
+    public void moduloOp(
```

Comment:
`NumericComparisonTypeCoercionTest` have verification that no one pair is missed even if new pair will be added to the enum (see `NumericComparisonTypeCoercionTest#argsIncludesAllTypePairs`). Do you have similar verification?